### PR TITLE
ZBUG-2368: ProxyServlet domain check: substring match vulnerability

### PR DESCRIPTION
**Pull request by thefoolwhocodes**
**Problem**
- Vulnerability while domain match

**Testing**
- Test for the use cases mentioned in ticket.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/zimlet/ProxyServlet.java [hunk 1]
```diff
@@ -99,22 +99,22 @@
...
+            ZimbraLog.zimlet.info("error getting allowedDomains: " + se.getMessage());
             return false;
         }
         for (String domain : domains) {
-            if (domain.equals("*")) {
-                return true;
-            }
             if (domain.charAt(0) == '*') {
                 domain = domain.substring(1);
+                if (host.endsWith(domain)) {
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/zimlet/ProxyServlet.java:113]: what if there is vulnerable domain for proxy with wildcard, should we handle it? In that case we will still allow proxy for allowed domain list.

Comment (thefoolwhocodes) [reply store/src/java/com/zimbra/cs/zimlet/ProxyServlet.java:113]: Can be controlled via allowed domain configuration.

**Review by girishbhamare — APPROVED** [review]
LGTM

**Review by abhishekdeshmukh2209 — APPROVED** [review]
LGTM

**Review by ronstra-synacor — APPROVED** [review]
LGTM as long as the administrators know they should use _both_ "zimbra.com" and "*.zimbra.com" if they want zimbra.com and its legitimate subdomains, and not just "*zimbra.com" which will match "foozimbra.com" etc. as the bug suggests.

**Pull request merged**